webapp/channels/src/components/post_view/post_list/post_list.tsx

```diff
@@ -366,10 +366,7 @@ export default class PostList extends React.PureComponent<Props, State> {
         }
 
         return (
-            <div
-                className='post-list-holder-by-time'
-                key={'postlist-' + this.props.channelId}
-            >
+            <div className='post-list-holder-by-time'>
```

Comment:
Of course, I don't know the code in mattermost. But it seems the problem isn't with the keys, but with the storage of the links to the DOM. They're left somewhere and aren't cleared. More likely, we need to look for refs whose remain active after switching channels.

Reply:
No, the detached VDOMs are those `<div className='post-list-holder-by-time'>` elements, see screenshots above or in the related issue